server/room.mjs

```diff
@@ -279,12 +281,20 @@ export default class Room {
   }
 
   setState(state) {
+    this.trace('state', 'setting state');
     const meta = this.state._meta;
     this.state = state;
     this.state._meta = meta;
     this.broadcast('state', state);
   }
 
+  trace(type, payload) {
+    if(this.traceActivated) {
+      const line = `${new Date().toISOString()} - ${type} - ${JSON.stringify(payload)}\n`;
+      fs.appendFileSync(path.resolve() + '/save/' + this.id + '.trace', line);
```

Comment:
Consider sanitizing `this.id` using something like this: 

https://www.npmjs.com/package/string-sanitizer

Reply:
In the meantime, server.mjs takes care of only accepting `/^[A-Za-z0-9_-]+$/` as room IDs.